crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -38,6 +51,182 @@ struct QueryLimitsCheckerExt {
+                                    .entry(arg.1.node.to_string())
+                                    .or_insert(tx_bytes_len);
+                            }
+                        }
+                    } else if f.node.name.node == EXECUTE_TX_BLOCK {
+                        let tx_bytes_len = f.node.arguments.iter().fold(0, |acc, arg| {
+                            acc + get_value_str_len(&arg.1.node, self.variables)
+                        });
+                        if tx_bytes_len > self.max_tx_payload_size as usize {
+                            self.log_metric(
+                                ctx,
+                                "The txBytes+signatures size of {} node is too large: {}",
+                                payload_size,
+                            );
+                            return Err(graphql_error_at_pos(
+                                    code::BAD_USER_INPUT,
+                                    format!(
+                                        "The txBytes+signatures size of {} node is too large. The maximum allowed is {} bytes",
+                                        f.node.name.node,
+                                        self.max_tx_payload_size
+                                    ),
+                                    f.pos
+                                ));
+                        }
+                        let var_name = f
+                            .node
+                            .arguments
+                            .iter()
+                            .map(|x| x.1.node.to_string())
+                            .collect::<Vec<_>>()
+                            .concat();
+                        // the key is the txBytes+signatures value itself or the variables' names
+                        track_vars.entry(var_name).or_insert(tx_bytes_len);
+                    }
+                }
+            }
+        }
+
+        let cfg = ctx
+            .data::<ServiceConfig>()
+            .expect("No service config provided in schema data");
+
+        // remove the executeTransactionBlock and the dryRunTransactionBlock variables from the
+        // total query size, and check if the rest is bigger than the allowed limit
+        if (payload_size - track_vars.values().sum::<usize>() as u64)
+            > cfg.limits.max_query_payload_size.into()
+        {
+            self.log_metric(ctx, "Query payload is too large: {}", payload_size);
+            return Err(graphql_error(
+                code::BAD_USER_INPUT,
+                format!(
+                    "Query payload is too large. The maximum allowed is {} bytes",
+                    cfg.limits.max_query_payload_size
+                ),
+            ));
+        }
+
+        Ok(())
+    }
+
+    /// Check the body content-length against the allowed payload. The content-length will include
+    /// variables data as well.
+    fn check_overall_payload_size(&self, ctx: &ExtensionContext<'_>) -> ServerResult<()> {
```

Comment:
I think with the earlier suggestion of making the tx bytes check cumulative we will no longer need this check.